lib/Common/Memory/HeapBlockMap.inl

```diff
@@ -142,13 +142,38 @@ HeapBlockMap32::Mark(void * candidate, MarkContext * markContext)
 #endif
         ((SmallFinalizableHeapBlock*)chunk->map[id2])->ProcessMarkedObject<doSpecialMark>(candidate, markContext);
         break;
+#ifdef RECYCLER_VISITED_HOST
```

Comment:
For visited objects, is mark always called with interior marking mode- if so, do we actually need to implement this or should we just assert/fail fast here?

Reply:
We do need to implement it.  This is the core path.  The interior template parameter of MarkingContext::Mark controls if we are using HeapBlockMap*::MarkInterior or HeapBlockMap*::Mark.  The interior template parameter though I believe is only true for the MemGC recycler, so the JS GC is always using HeapBlockMap*::Mark.

Inside HeapBlockMap*::Mark we must still support having interior pointers to the new types of blocks we added, since we expect the majority of our pointers to be interior due to our wrapper approach for implementing IRecyclerVisitedObject in the host.  As a result, once we know the block type in the HeapBlockMap*::Mark method, we can support interior references by calling MarkInteriorInternal on a computed pointer obtained from GetRealAddressFromInterior.

Reply:
I think I could assert/fail in the markinterior version.  I'll try adding that.

Reply:
The other place which sets enableScanInteriorPointers (which leads to calling HeapBlockMap*::MarkInterior) is in the Recycler constructor based on a CUSTOM_CONFIG_FLAG.  If I force failure under the MarkInterior path then allocating our new type of heap block would lead to a failure in that configuration.  Is that OK or should MarkInterior stay as is?